### PR TITLE
Raise LexException when codec can't decode some bytes

### PR DESCRIPTION
**Pull request by waigx**
Fixes https://github.com/hylang/hy/issues/1571

### CODE REVIEW

File: tests/test_lex.py [hunk 1]
```diff
@@ -69,6 +69,12 @@ def test_lex_strings():
     assert objs == [HyString("abc")]
 
 
+def test_lex_strings_exception():
+    """ Make sure tokenize throws when codec can't decode some bytes"""
+    with lexe() as e: tokenize('\"\\x8\"')
```
Comment (vodik) [inline tests/test_lex.py]: Do this across two lines *cough* *cough* PEP8 :wink: 

That said, I'd suggest renaming `e` to `execinfo`. Its consistent with other unit tests, and I think also better names what's provided as it isn't the exception itself.

Comment (waigx) [reply tests/test_lex.py]: Ah, sorry I was just copying and pasting other part of this file...

Really should have a good read of PEP8, and maybe find some PEP8 plugin for vim...

File: tests/test_lex.py [hunk 2]
```diff
@@ -69,6 +69,13 @@ def test_lex_strings():
     assert objs == [HyString("abc")]
 
 
+def test_lex_strings_exception():
+    """ Make sure tokenize throws when codec can't decode some bytes"""
+    with lexe() as execinfo:
+        tokenize('\"\\x8\"')
+    assert "Can't convert \"\\x8\" to a HyString" in str(e.value)
```
Comment (vodik) [inline tests/test_lex.py]: Forget to update `e.value`? :wink:

Comment (waigx) [reply tests/test_lex.py]: D'OH

**Comment by Kodiologist** [comment]
N.B. GitHub won't detect "Fix issue #1571" properly; [you need the issue number right after the word "fix"](https://help.github.com/articles/closing-issues-using-keywords/).

**Comment by waigx** [comment]
@Kodiologist - Thanks, good to know that. I just just updated the initial comment.

File: tests/test_lex.py (hunk 1 above)
Comment (Kodiologist) [reply tests/test_lex.py]: @waigx You can use `make flake` to run flake8 on the Hy codebase.

**Comment by Kodiologist** [comment]
Looks good.

1. Don't you mean "**Throw** when codec can't decode some bytes"?
2. Update NEWS, unless you know this to be a regression since the previous release.

**Comment by waigx** [comment]
@Kodiologist - Thanks, just added the NEWS (shamefully forgot that again... ) and re-worded the title. Please take another look.

**Pull request merged**